src/wallet/view_seed_flow.c

```diff
@@ -0,0 +1,205 @@
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "constant_texts.h"
+#include "core_error.h"
+#include "reconstruct_wallet_flow.h"
+#include "ui_delay.h"
+#include "ui_list.h"
+#include "ui_message.h"
+#include "ui_multi_instruction.h"
+#include "ui_state_machine.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef enum view_seed_states {
+  VIEW_SEED_INSTRUCTIONS = 1,
+  VIEW_SEED_MNEMONIC_RECONSTRUCTION,
+  VIEW_SEED_RECONSTRUCT_SUCCESS_MESSAGE,
+  VIEW_SEED_MNEMONIC_DISPLAY,
+  VIEW_SEED_COMPLETED,
+  VIEW_SEED_COMPLETED_WITH_ERRORS,
+  VIEW_SEED_TIMED_OUT,
+  VIEW_SEED_EARLY_EXIT,
+  VIEW_SEED_EXIT,
+} view_seed_states_e;
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief The function handles different states and transitions of view seed
+ * flow based on the current state.
+ *
+ * @param current_state The current state of the view seed process. It is of
+ * type `view_seed_states_e`, which is an enumeration representing different
+ * states of the process.
+ * @param wallet_id A pointer to a uint8_t array representing the wallet ID.
+ *
+ * @return the next state of the view_seed_states_e enum type.
+ */
+view_seed_states_e view_seed_handler(view_seed_states_e current_state,
+                                     const uint8_t *wallet_id);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+CONFIDENTIAL char mnemonics[MAX_NUMBER_OF_MNEMONIC_WORDS]
```

Comment:
if we can remove the states/switch cases, this global variable can be dissolved.

Reply:
refactored